### PR TITLE
New package xchain-cosmos-sdk

### PR DESCRIPTION
**Pull request by hippocampus-web3**
**Description:**

This PR introduces a new package, `xchain-cosmos-sdk`, aimed at consolidating common logic for packages utilizing CosmosSDK technology within the Xchainjs repository as the existing basic packages containing common logic for UTXO and EVM chains.

**Objectives:**

_Unified Support:_ The primary goal is to offer unified support for cryptocurrency wallets across different chains.

**Implementation Details:**

_Official Dependencies:_ The official dependencies provided by CosmosSDK, the @cosmjs package set, are utilized to interact with nodes from JavaScript.

_Base Package Creation:_ This initial PR focuses on creating the base package, with subsequent tasks planned to extend functionality and support.

**Future Work:**

_Kujira Wallet Support:_ The first task post this PR is to extend support to the Kujira wallet to test the new implementation.

_Update Existing Wallets:_ Subsequently, each wallet supported by Xchain that works with CosmosSDK will be updated to avoid the current code duplication (cosmos, thorchain, binance, and maya).

This package is crucial to avoid code duplication and to ensure more coherent and efficient support for wallets working with CosmosSDK technology in the Xchain ecosystem.

**ADDITIONAL NOTES**:

**Performance Optimization:** 
Given the number of wallets supported by Xchain and the computational cost of some address generation processes, I thnk it is crucial to convert address generation into an asynchronous process to avoid browser thread blocking and freezing.

**Rationale:**

The getAddress method of the XChainClient interface remains synchronous in this PR, with addresses being generated manually instead of using @cosmjs. This might be a significant change but we should start thinking about make the address generation asynchronous

This future change will help to optimize performance and prevent browser thread blocking, which is common in projects using Xchain that generate all address together at the start of the application due to the computational cost of some address generation processes.

The asynchronous approach to address generation is pivotal to enhancing user experience by preventing browser blockages and improving overall performance.

### CODE REVIEW

File: packages/xchain-cosmos-sdk/package.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    "lib"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:xchainjs/xchainjs-lib.git"
+  },
+  "scripts": {
+    "clean": "rimraf lib/**",
+    "build": "yarn clean && rollup -c",
+    "test": "jest --passWithNoTests",
```
Comment (Thorian1te) [inline packages/xchain-cosmos-sdk/package.json:29]: add `  "e2e": "jest --config jest.config.e2e.js",` for terminal support.

Comment (hippocampus-web3) [reply packages/xchain-cosmos-sdk/package.json:29]: Done

**Comment by Thorian1te** [comment]
Just quickly, do any packages depending on the new `xchain-utils` version need bumping?

**Comment by hippocampus-web3** [comment]
Not bumping necessary. The change on xchain-utils is a minor and retrocompatible change

**Review by Thorian1te — APPROVED** [review]

**Pull request merged**